workloads/vectorindex-controller/vectorindex-controller.py

```diff
@@ -0,0 +1,128 @@
+import os
+from tempfile import NamedTemporaryFile
+
+import numpy as np
+
+from tqdm import tqdm
+
+from kubernetes import config, client, watch
+from prometheus_client import Counter, start_http_server
+
+from ocifs import OCIFileSystem
+
+import pyarrow.feather
+
+from sentence_transformers import SentenceTransformer
```

Comment:
I don't have a strong preference for single-use situations like this. I'm pretty sure that I copied both the `pyarrow` and `sentence_transformers` imports from upstream example code, which is why they differ.

If I need to use a name many times, then I prefer the a-la-carte imported names with `from` because it leads to shorter code. I'll make exceptions for modules like `json` whose exported names are too short and easily shadowed.